### PR TITLE
Add support of NFT(ERC721) Non-Fungible Token Standard

### PR DESCRIPTION
**Pull request by MaximFischuk**

### CODE REVIEW

File: src/main/java/io/zksync/domain/token/NFT.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    private String creatorId;
+
+    private String contentHash;
+
+    @Override
+    public BigDecimal intoDecimal(BigInteger amount) {
+        return new BigDecimal(amount)
+            .setScale(1)
+            .divide(BigDecimal.TEN.pow(1), RoundingMode.DOWN);
```
Comment (StanislavBreadless) [inline src/main/java/io/zksync/domain/token/NFT.java:28]: Not sure what this operation does

File: src/main/java/io/zksync/signer/SigningUtils.java [hunk 2]
```diff
@@ -15,7 +16,7 @@
 
 public class SigningUtils {
 
-    private static final int MAX_NUMBER_OF_TOKENS = 128;
+    private static final long MAX_NUMBER_OF_TOKENS = Long.MAX_VALUE;
```
Comment (StanislavBreadless) [inline src/main/java/io/zksync/signer/SigningUtils.java:19]: It should be `unsigned 2^32`

**Review by StanislavBreadless — COMMENTED** [review]
:+1: 

One more question

File: src/main/java/io/zksync/signer/SigningUtils.java (hunk 2 above)
Comment (MaximFischuk) [reply src/main/java/io/zksync/signer/SigningUtils.java:19]: Unfortunately Java has no any unsigned type thus actually uses greater types to cover set of unsigned values. As alternative I can add invariant checking within convertion into bytes

File: src/main/java/io/zksync/domain/token/NFT.java (hunk 1 above)
Comment (MaximFischuk) [reply src/main/java/io/zksync/domain/token/NFT.java:28]: This method converts amount of NFT (always 1) to decimals representation. As I understand ZkSync uses 1 decimals for NFT

Comment (StanislavBreadless) [reply src/main/java/io/zksync/domain/token/NFT.java:28]: We should not, I have checked and we indeed use 1 for some reason. This is strange. I will talk to the team

**Pull request merged**